packages/edit-site/src/components/block-editor/index.js

```diff
@@ -149,20 +171,77 @@ export default function BlockEditor() {
 		usePageContentLockNotifications(),
 	] );
 	const isMobileViewport = useViewportMatch( 'small', '<' );
-	const { clearSelectedBlock } = useDispatch( blockEditorStore );
+	const { clearSelectedBlock, selectBlock } = useDispatch( blockEditorStore );
 	const [ resizeObserver, sizes ] = useResizeObserver();
 
-	const isTemplatePart = templateType === 'wp_template_part';
+	const isFocusMode = FOCUSABLE_ENTITIES.includes( templateType );
 
 	const hasBlocks = blocks.length !== 0;
 	const enableResizing =
-		isTemplatePart &&
+		isFocusMode &&
 		canvasMode !== 'view' &&
 		// Disable resizing in mobile viewport.
 		! isMobileViewport;
 	const isViewMode = canvasMode === 'view';
+	const isEditMode = canvasMode === 'edit';
+
+	const firstBlockClientId = blocks[ 0 ]?.clientId;
+
+	const isTemplateTypeNavigation = templateType === 'wp_navigation';
+
 	const showBlockAppender =
-		( isTemplatePart && hasBlocks ) || isViewMode ? false : undefined;
+		( isTemplateTypeNavigation && isFocusMode && hasBlocks ) || isViewMode
+			? false
+			: undefined;
+
+	useEffect( () => {
+		setBlockEditingMode(
+			firstBlockClientId,
+			isTemplateTypeNavigation ? 'contentOnly' : 'default'
+		);
+		return () => {
+			unsetBlockEditingMode( firstBlockClientId );
+		};
+	}, [
+		firstBlockClientId,
+		isTemplateTypeNavigation,
+		unsetBlockEditingMode,
+		setBlockEditingMode,
+	] );
+
+	useEffect( () => {
+		if ( isEditMode && isTemplateTypeNavigation ) {
+			selectBlock( firstBlockClientId );
+		}
+	}, [
+		firstBlockClientId,
+		isEditMode,
+		isTemplateTypeNavigation,
+		selectBlock,
+	] );
+
+	useEffect( () => {
+		// Disabled global inserter in Navigation focus mode
+		// when no block is selected to avoid ability to
+		// insert blocks outside of the Navigation block.
+		if ( ! isTemplateTypeNavigation ) {
+			return;
+		}
+
+		if ( !! selectedBlockClientId ) {
+			setIsInserterDisabled( false );
+		} else {
+			setIsInserterDisabled( true );
+		}
+
+		return () => {
+			setIsInserterDisabled( false );
+		};
+	}, [
+		selectedBlockClientId,
+		setIsInserterDisabled,
+		isTemplateTypeNavigation,
+	] );
```

Comment:
@youknowriad Is there a better way to limit the editor to only allow a _single_ root block? 

Navigation focus mode is a special case because we are editing an entity (i.e. `wp_navigation` post) but the editor also contains a Navigation block itself.

Therefore we need to disabled the editor from being to insert any more blocks at the root level both programmatically and via the UI.

Reply:
Ultimately I used `templateLock` and `template` as the `settings` on the `BlockEditorProvider` which does what we want. Using `insert` as the lock mode seems to give us what we want but we need to stop that cascading by applying the `templateLock=false` attribute to the "ghost" Nav block at point of creation to unset the lock for the inner blocks.